.github/test-reporter.yml

```diff
@@ -0,0 +1,20 @@
+---
+## Workflow to process the JUnit test results and add a report to the checks.
+name: Test Report
+on:
+  workflow_run:
+    workflows:
+      - test
+    types:
+      - completed
+
+jobs:
+  report:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: elastic/apm-pipeline-library/.github/actions/test-report@gh_actions
+        with:
+          artifact: test-results            # artifact name
+          name: JUnit Tests                 # Name of the check run which will be created
+          path: "**/*.xml"                  # Path to test results (inside artifact .zip)
+          reporter: java-junit              # Format of test results
```

Comment:
Would it be possible to encapsulate the conversion to JUnit in the test-report action, so the Go tests could output in `go test -json` format (https://pkg.go.dev/cmd/test2json)?

In apm-server we're using `gotestsum` to do this: https://github.com/elastic/apm-server/blob/7b544a98c45cf6e3a2eae71f475a62a19d95e0ef/.ci/scripts/unit-test.sh#L15-L20

It would be possible to convert existing results by specifying a `--raw-command` to the gotestsum program, which would just `cat <results.json>`.

Reply:
> Would it be possible to encapsulate the conversion to JUnit in the test-report action, so the Go tests could output in go test -json format (https://pkg.go.dev/cmd/test2json)?

I think so,  https://github.com/elastic/apm-pipeline-library/blob/main/.github/actions/test-report/action.yml is the action we encapsulate, https://github.com/dorny/test-reporter/issues/189 has not been implemented but we can do the trick in our end, as we can add support for a new `reporter`, and do the magic


If you don't mind I'll work on supporting this particular scenario in this PR and see how it goes

Reply:
For now, I'll not enable the test-reporting as we wanna make the contribution to the upstream project, so I'll keep it this simpler for now